downloads/swan-lake-release-notes/swan-lake-2201.4.0/RELEASE_NOTE.md

```diff
@@ -88,6 +88,12 @@ To view bug fixes, see the [GitHub milestone for 2201.4.0 (Swan Lake)](https://g
 ### Improvements
 
 #### OpenAPI Tool
+- Added mapping for the `additionalProperty` property in the OpenAPI object schema. With this support, generated Ballerina records can be open records or closed records according to the `additionalProperty` property details.
+  Thereby, some of your existing generated Ballerina records may change into close records after regenerating using Ballerina 2201.4.0 (Swan Lake) version.
+- Improved OpenAPI schema `nullable:true` property support by providing generated record field with default of value nil, instead of making field both nilable and optional.
+  Therefore, some of your existing generated Ballerina records field may change into default value nil using Ballerina 2201.4.0 (Swan Lake) version.
+- Changed generated client remote/resource methods' request body and return types, `json` into `http:Request` and `http:Response` types for unhandled Request, Response payload media types in OpenAPI specification.
```

Comment:
```suggestion
- Changed the request body and return types of the generated client remote/resource methods (from `json` to `http:Request` and `http:Response` types) for unhandled `Request`, and `Response` payload media types in the OpenAPI specification.
```
Check if the technical meaning is affected by the suggestions.

Reply:
@lnash94 Is this done only for the service generation side?

Reply:
This is done for client generation only. Service generation is yet to be done.